core-bundle/src/Controller/ContentElement/TemplateController.php

```diff
@@ -14,25 +14,28 @@
 
 use Contao\ContentModel;
 use Contao\CoreBundle\DependencyInjection\Attribute\AsContentElement;
+use Contao\CoreBundle\Twig\FragmentTemplate;
 use Contao\StringUtil;
-use Contao\Template;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 
 #[AsContentElement(category: 'includes')]
 class TemplateController extends AbstractContentElementController
 {
-    protected function getResponse(Template $template, ContentModel $model, Request $request): Response
+    protected function getResponse(FragmentTemplate $template, ContentModel $model, Request $request): Response
     {
         $data = StringUtil::deserialize($model->data, true);
 
-        $template->keys = array_combine(
-            array_column($data, 'key'),
-            array_column($data, 'value')
+        $template->set(
+            'keys',
+            array_combine(
+                array_column($data, 'key'),
+                array_column($data, 'value')
+            )
         );
 
-        // Backwards compatibililty
-        $template->data = $data;
+        // Backwards compatibility
+        $template->set('data', $data);
```

Comment:
Do we still need this backwards compatibility?

Reply:
I wasn't sure about this tbh. IIRC we intentionally left it in for 5.0 but then again we did not think about making the templates Twig only back then. Shall we remove it?

Reply:
I’d remove it as the templates are not backwards compatible anyways.

Reply:
Done in b67226d